### PR TITLE
Better unchecked warnings.

### PR DESCRIPTION
**Pull request by paulp**

### CODE REVIEW

**Comment by paulp** [comment]
PLS REBUILD ALL

**Comment by paulp** [comment]
Review by @adriaanm since he seems to be all back and stuff.  Tell me I missed some massively easier way.

File: src/compiler/scala/tools/nsc/typechecker/Checkable.scala [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+      val outcomes = f"""
+        |[P1] $P1%-6s X <: P             // $X  <: $P
+        |[P2] $P2%-6s X !<: P            // $X !<: $P for all X, P
+        |[P3] $P3%-6s XR <: P            // $XR <: $P
+        |[P4] $P4%-6s None of the above  // !(P1 || P2 || P3)
+      """.stripMargin.trim
+
+      val msg = List(facts, outcomes) mkString ("\n", "\n", "\n")
+      log(msg)
```
Comment (retronym) [inline src/compiler/scala/tools/nsc/typechecker/Checkable.scala]: Create the message inside the lazy arg to `log`.

**Comment by gkossakowski** [comment]
Just to set right expectations on @adriaanm's availability: he is back but rather busy with non-work stuff so I wouldn't put too much pressure on him.

Other than that it looks like a fairly big patch. I hope it can get lgtm by Friday.

**Comment by retronym** [comment]
To make this a bit easier to review, could please you split the commit into two pieces: one that extracts the `Checkable` trait, and a second that implements the behavioural changes?

**Comment by paulp** [comment]
Okay, forget @adriaanm, I'll attempt to meet @retronym's review demands.

File: src/compiler/scala/tools/nsc/typechecker/Checkable.scala (hunk 1 above)
Comment (paulp) [inline src/compiler/scala/tools/nsc/typechecker/Checkable.scala]: Ugh, oops. I was trying to get this in before I had to bail yesterday.

**Comment by jsuereth** [comment]
BTW - We had a discussion at the Scala meeting on releasing.  I think this is best suited to target 2.10.1 (which should be out in 4 months + N weeks).   We'll lock the date for 2.10.1 down soon.

Also, we haven't firmed up where 2.10.1 dev will happen yet.   I'll get back to you on whether we spin up a temporary branch for the next month, or just put things in master.

**Comment by paulp** [comment]
You should turn off -unchecked by default if this isn't going in.

**Comment by retronym** [comment]
I would like to see this one given a chance. If it issues a spurious warning that needs be refined in RC 2 or 3, what's the harm? It won't block people using the RC for other means.

**Comment by paulp** [comment]
I have made the patch much stronger.

**Comment by paulp** [comment]
PLS REBUILD ALL

**Comment by paulp** [comment]
files/run/t6344.scala is going to fail because that checkfile update wasn't in.  If that's it, let's pretend it passed.

**Comment by paulp** [comment]
Indeed:

```
build.xml:2649: java.lang.RuntimeException: Test suite finished with 1 case failing:
test/files/run/t6344.scala [FAILED]
```

**Comment by paulp** [comment]
Reopening a new one.

**Pull request closed without merging**